3rd_party/3rd_party.cmake

```
@@ -1,274 +1,282 @@
-      foreach(RES ${FIND_RES})
-        # Replace RPATH for 3rd party libraries that already have one
-        execute_process(COMMAND patchelf --print-rpath ${RES} COMMAND grep lib OUTPUT_VARIABLE RPATH_VAR ERROR_VARIABLE RPATH_ERR WORKING_DIRECTORY "${INSTALL_DIR}" OUTPUT_STRIP_TRAILING_WHITESPACE)
-        if(RPATH_VAR)
-          execute_process(COMMAND patchelf --force-rpath --set-rpath '$ORIGIN' ${RES} OUTPUT_VARIABLE SET_RPATH_OUT ERROR_VARIABLE SET_RPATH_ERR WORKING_DIRECTORY "${INSTALL_DIR}" OUTPUT_STRIP_TRAILING_WHITESPACE)
-          if(NOT SET_RPATH_ERR)
-            message(STATUS "Set RPATH in ${RES}: ${SET_RPATH_OUT}")
-          endif()
-        else()
-          message(STATUS "Did not set RPATH in ${RES}")
-        endif()
-      endforeach()
-    endif()
-  endif()
-
-endfunction()
-
-if(INSTALL_DIR)
-  third_party(add ${INSTALL_DIR})
-endif()
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+#
+
+#
+# Various other C programs/libraries need to be installed on the local machine,
+# in directories listed in the platform specific variables listed below.
+#
+# Usage:
+#  cmake -D INSTALL_DIR=${INSTALL_DIR} -P 3rd_party.cmake
+#
+# A complication is that Java's zip functionality turns symlinks into actual
+# files, so we need to ensure libraries are only picked up by the name that the
+# dynamic loader will look for, and no symlinks are used.  Otherwise the
+# installer becomes very bloated.
+#
+string(TOLOWER ${CMAKE_HOST_SYSTEM_NAME} HOST_SYSTEM_NAME)
+message(STATUS "3rd_party: HOST_SYSTEM_NAME=${HOST_SYSTEM_NAME}")
+
+if ("${HOST_SYSTEM_NAME}" STREQUAL "windows")
+  # We only support x86_64
+  set(HOST_SYSTEM_PROCESSOR "x86_64")
+else()
+  execute_process(COMMAND uname -m OUTPUT_VARIABLE HOST_SYSTEM_PROCESSOR OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE arm aarch HOST_SYSTEM_PROCESSOR ${HOST_SYSTEM_PROCESSOR})
+endif()
+message(STATUS "3rd_party: HOST_SYSTEM_PROCESSOR=${HOST_SYSTEM_PROCESSOR}")
+set(CMAKE_TOOLCHAIN_FILE "cmake/${HOST_SYSTEM_NAME}-${HOST_SYSTEM_PROCESSOR}.cmake")
+
+set(ARCH ${HOST_SYSTEM_PROCESSOR})
+set(CMAKE_SYSTEM_NAME ${HOST_SYSTEM_NAME})
+
+if ("${CMAKE_SYSTEM_NAME}" STREQUAL "darwin")
+  message(STATUS "3rd_party: CMAKE_SYSTEM_NAME = ${CMAKE_SYSTEM_NAME}")
+  message(STATUS "3rd_party: Copying macOS 3rd party libraries")
```

nit: in all the other branches these messages are the other way around

```suggestion
  message(STATUS "3rd_party: Copying macOS 3rd party libraries")
  message(STATUS "3rd_party: CMAKE_SYSTEM_NAME = ${CMAKE_SYSTEM_NAME}")
```